controller/eventQuestionController.js

```diff
@@ -0,0 +1,291 @@
+
+const create = (req, res) => {
+
+    const eventQuestion = new EventQuestion(req.body);
+
+    eventQuestion.save().then(() => {
+        return res.status(HttpStatus.CREATED).json(eventQuestion);
+    }).catch(err => {
+        const errorMessage = ValidationUtil.buildErrorMessage(err, 'create', 'eventQuestion');
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: errorMessage
+        });
+    });
+
+};
+
+const getById = (req, res) => {
+
+    const eventQuestionId = req.params.id;
+    const user = req.decoded;
+
+    if (!ValidationUtil.isValidObjectId(eventQuestionId)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+
+    if (!ValidationUtil.isUserAdmin(user.accessType)) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized'
+        });
+    }
+
+    EventQuestion.findById(eventQuestionId, { _id: 0 }).then(eventQuestion => {
+        if (!eventQuestion) {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event question not found'
+            });
+        } else {
+            Event.findOne({ _id: new Mongoose.Types.ObjectId(eventQuestion.eventID), companyID: new Mongoose.Types.ObjectId(user.companyID) }, { _id: 0 }).then(event => {
+                if (event) {
+                    return res.status(HttpStatus.OK).json(eventQuestion);
+                } else {
+                    return res.status(HttpStatus.NOT_FOUND).json({
+                        status: StatusEnum['ERROR'],
+                        message: 'Event not found'
+                    });
+                }
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: 'Internal server error'
+        });
+    });
+
+}
+
+const deleteById = (req, res) => {
+
+    const eventQuestionId = req.params.id;
+    const user = req.decoded;
+
+    if (!ValidationUtil.isValidObjectId(eventQuestionId)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+
+    if (!ValidationUtil.isUserAdmin(user.accessType)) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized'
+        });
+    }
+
+    EventQuestion.findByIdAndDelete(eventQuestionId).then(eventQuestion => {
+        if (eventQuestion) {
+            // since it is a simple delete, should a boolean value be returned or the entire object => DISCUSS
+            return res.status(HttpStatus.OK).json(eventQuestion);
```

Comment:
Returning the entire object is okay because we do that for every successful CRUD operation.